src/util/notifications_util.js

```diff
@@ -514,10 +519,36 @@ function get_notification_logger(locking, namespace, poll_interval) {
         namespace = node_name + '_' + config.NOTIFICATION_LOG_NS;
     }
 
-    return new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
+    const logger = new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
         locking,
         poll_interval,
     });
+
+    //initialize writes_counter, used in check_free_space
+    logger.writes_counter = 0;
+
+    return logger;
+}
+
+//If space check is configures, create an event in case free space is below threshold.
+function check_free_space(req) {
+    if (!req.object_sdk.nsfs_config_root || !config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        //free space check is disabled. nothing to do.
+        return;
+    }
+    req.notification_logger.writes_counter += 1;
+
+    //is it time to check?
+    if (req.notification_logger.writes_counter > config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        //yes. remember we've just ran  the check by zero-ing the counter.
+        req.notification_logger.writes_counter = 0;
+        const fs_stat = fs.statfsSync(config.NOTIFICATION_LOG_DIR);
```

Comment:
I know that in NSFS we use: `await nb_native().fs.stat` to get the stat.
Why did you choose to use stat from fs and in its sync way?
I think it's worth adding a comment about it.